fido/Crypto/Fido2/Model/JavaScript/Decoding.hs

```diff
@@ -277,6 +284,123 @@ instance Decode (M.PublicKeyCredential 'M.Get) where
     pkcClientExtensionResults <- decode clientExtensionResults
     pure $ M.PublicKeyCredential {..}
 
+instance Decode M.RpId
+
+instance Decode M.RelyingPartyName
+
+instance Decode M.PublicKeyCredentialRpEntity where
+  decode JS.PublicKeyCredentialRpEntity {..} = do
+    pkcreId <- decode id
+    pkcreName <- decode name
+    pure $ M.PublicKeyCredentialRpEntity {..}
+
+instance Decode M.UserAccountDisplayName
+
+instance Decode M.UserAccountName
+
+instance Decode M.PublicKeyCredentialUserEntity where
+  decode JS.PublicKeyCredentialUserEntity {..} = do
+    pkcueId <- decode id
+    pkcueDisplayName <- decode displayName
+    pkcueName <- decode name
+    pure $ M.PublicKeyCredentialUserEntity {..}
+
+instance Decode M.Challenge
+
+instance Decode PublicKey.COSEAlgorithmIdentifier where
+  decode n = maybe (Left $ DecodingErrorUnexpectedAlgorithmIdentifier n) Right $ PublicKey.toAlg n
+
+instance Decode M.Timeout
+
+instance Decode [M.AuthenticatorTransport] where
+  decode = pure . mapMaybe decodeTransport
+    where
+      decodeTransport "usb" = Just M.AuthenticatorTransportUSB
+      decodeTransport "nfc" = Just M.AuthenticatorTransportNFC
+      decodeTransport "ble" = Just M.AuthenticatorTransportBLE
+      decodeTransport "internal" = Just M.AuthenticatorTransportInternal
+      decodeTransport _ = Nothing
```

Comment:
All of these special cases should also have a link to the spec and an explanation